classes/local/store/s3/client.php

```diff
@@ -244,100 +275,102 @@ protected function get_filepath_from_hash($contenthash) {
-                'Body' => 'test content'));
+                'Body' => 'test content',
+            ]);
         } catch (\Aws\S3\Exception\S3Exception $e) {
-            $details = $this->get_exception_details($e);
-            $permissions->messages[get_string('settings:writefailure', 'tool_objectfs') . $details] = 'notifyproblem';
-            $permissions->success = false;
+            $exdetails = $this->get_exception_details($e);
+            $details .= get_string('settings:writefailure', 'tool_objectfs') . $exdetails;
+            $status = result::ERROR;
         }
 
+        // Try to get an object.
         try {
-            $result = $this->client->getObject(array(
+            $this->client->getObject([
                 'Bucket' => $this->bucket,
-                'Key' => $this->bucketkeyprefix . 'permissions_check_file'));
+                'Key' => $this->bucketkeyprefix . 'permissions_check_file',
+            ]);
         } catch (\Aws\S3\Exception\S3Exception $e) {
             $errorcode = $e->getAwsErrorCode();
             // Write could have failed.
             if ($errorcode !== 'NoSuchKey') {
-                $details = $this->get_exception_details($e);
-                $permissions->messages[get_string('settings:permissionreadfailure', 'tool_objectfs') . $details] = 'notifyproblem';
-                $permissions->success = false;
+                $exdetails = $this->get_exception_details($e);
+                $details .= get_string('settings:permissionreadfailure', 'tool_objectfs') . $exdetails;
+                $status = result::ERROR;
             }
         }
 
+        // Try to delete an object.
         if ($testdelete) {
             try {
-                $result = $this->client->deleteObject(array('Bucket' => $this->bucket, 'Key' => $this->bucketkeyprefix . 'permissions_check_file'));
-                $permissions->messages[get_string('settings:deletesuccess', 'tool_objectfs')] = 'warning';
-                $permissions->success = false;
+                $this->client->deleteObject([
+                    'Bucket' => $this->bucket,
+                    'Key' => $this->bucketkeyprefix . 'permissions_check_file',
+                ]);
+                $details .= get_string('settings:deletesuccess', 'tool_objectfs');
             } catch (\Aws\S3\Exception\S3Exception $e) {
                 $errorcode = $e->getAwsErrorCode();
                 // Something else went wrong.
                 if ($errorcode !== 'AccessDenied') {
                     $details = $this->get_exception_details($e);
-                    $permissions->messages[get_string('settings:deleteerror', 'tool_objectfs') . $details] = 'notifyproblem';
-                    $permissions->success = false;
+                    $details .= get_string('settings:deleteerror', 'tool_objectfs') . $details;
+                    $status = result::ERROR;
                 }
             }
         }
```

Comment:
Ideally revert this back to previous and just implode the array of messages at the end - a bit simpler and less diff